common/src/main/java/dev/ftb/mods/ftblibrary/sidebar/SidebarGroupGuiButton.java

```diff
@@ -11,152 +14,499 @@
+
+			if (mx >= addIconX && my >= addIconY && mx < addIconX + 16 && my < addIconY + 16) {
+				isMouseOverAdd = true;
+				Color4I.WHITE.withAlpha(137).draw(graphics, addIconX + 1, addIconY + 1, 16, 16);
+			}
+
+			if (addBoxOpen) {
+				int maxWidth = 0;
+				for (SidebarGuiButton button : disabledButtonList) {
+					String s = I18n.get(button.getSidebarButton().getLangKey());
+					int width = Minecraft.getInstance().font.width(s);
+					maxWidth = Math.max(maxWidth, width);
+				}
+
+				int gridY = gridStartBottom ? addIconY - disabledButtonList.size() * BUTTON_SPACING : BUTTON_SPACING;
+				int gridX = gridStartRight ? addIconX - maxWidth - 6 : (currentGirdWidth + 1) * BUTTON_SPACING;
+
+				graphics.pose().pushPose();
+				graphics.pose().translate(0, 0, 1000);
+				if (gridStartRight) {
+					drawHoveredGrid(graphics, addIconX, gridY, 1, disabledButtonList.size(), BUTTON_SPACING, Color4I.GRAY, Color4I.BLACK, mx, my, gridStartBottom, gridStartRight, gridX, gridY);
+					drawGrid(graphics, addIconX - maxWidth - 6, gridY, 1, disabledButtonList.size(), maxWidth + 6, BUTTON_SPACING, Color4I.GRAY, Color4I.BLACK);
+				} else {
+					drawHoveredGrid(graphics, gridX, gridY, 1, disabledButtonList.size(), BUTTON_SPACING, Color4I.GRAY, Color4I.BLACK, mx, my, gridStartBottom, gridStartRight, gridX, gridY);
+					drawGrid(graphics, gridX + BUTTON_SPACING, gridY, 1, disabledButtonList.size(), maxWidth + 6, BUTTON_SPACING, Color4I.GRAY, Color4I.BLACK);
+				}
 
-		setX(Integer.MAX_VALUE);
-		setY(Integer.MAX_VALUE);
-		var maxX = Integer.MIN_VALUE;
-		var maxY = Integer.MIN_VALUE;
+				for (int i = 0; i < disabledButtonList.size(); i++) {
+					SidebarGuiButton button = disabledButtonList.get(i);
+					if (selectedButton != null && selectedButton == button) {
+						continue;
+					}
+					int buttonY = gridY + BUTTON_SPACING * i;
+					button.x = gridStartRight ? addIconX : gridX;
+					button.y = buttonY;
+					GuiHelper.setupDrawing();
+
+					if (mx >= button.x && my >= button.y && mx < button.x + 16 && my < button.y + 16) {
+						Color4I.WHITE.withAlpha(137).draw(graphics, button.x + 1, button.y + 1, 16, 16);
+						mouseOver = button;
+					}
+
+					button.getSidebarButton().getData().icon().draw(graphics, button.x + 1, button.y + 1, 16, 16);
+
+					String langText = I18n.get(button.getSidebarButton().getLangKey());
+					int textXPos = gridStartRight ? addIconX - Minecraft.getInstance().font.width(langText) - 2 : gridX + BUTTON_SPACING + 3;
+					graphics.drawString(Minecraft.getInstance().font, langText, textXPos, buttonY + 5, 0xFFFFFFFF);
+				}
+				graphics.pose().popPose();
 
-		for (var b : buttons) {
-			if (b.x >= 0 && b.y >= 0) {
-				setX(Math.min(getX(), b.x));
-				setY(Math.min(getY(), b.y));
-				maxX = Math.max(maxX, b.x + 16);
-				maxY = Math.max(maxY, b.y + 16);
 			}
+		}
+	}
 
-			if (mx >= b.x && my >= b.y && mx < b.x + 16 && my < b.y + 16) {
-				mouseOver = b;
+	@Override
+	public void onRelease(double d, double e) {
```

Comment:
This method has a bunch of unneeded nesting, please return early when you can and avoid the nesting where possible. This method also has a lot going on, try and see if you can pull some of the logic out into methods where possible